independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/BootstrapAppModelFactory.java

```diff
@@ -284,7 +296,7 @@ public CurationResult resolveAppModel() throws BootstrapException {
                 }
             }
             CurationResult curationResult = new CurationResult(getAppModelResolver()
-                    .resolveManagedModel(appArtifact, forcedDependencies, managingProject));
+                    .resolveManagedModel(appArtifact, forcedDependencies, managingProject, localArtifacts));
```

Comment:
I think the app model resolver should be providing the local artifacts instead of the other way around. It resolves them from the workspace anyway.

Reply:
Not every path in the resolver loads the workspace, in particular re-augmentation does not have a workspace available, and if we end up supporting upgrades it won't have it either.

Reply:
Why do we need to treat dependencies from the same workspace as the app root artifact differently than others? Dev and test (given that the jars might not be available yet) modes are obvious. Otherwise, it's all jars. If we do need to unpack certain jars for processing, it'd be cleaner to create a workspace impl for that, imo.

Reply:
They are candidates to be updated in remote dev mode, and so need to be treated differently when running in remote dev. 

Thinking about it though as these are part of the app if we package them in the lib dir with the rest of the dependencies then when they are updated the full lib layer of the docker image will be invalidated. These should also go in their own directory and have their own layer.